File: embedchain/loaders/local_qna_pair.py
from embedchain.helper_classes.json_serializable import register_deserializable
from embedchain.loaders.base_loader import BaseLoader


@register_deserializable
class LocalQnaPairLoader(BaseLoader):
    def load_data(self, content):
        """Load data from a local QnA pair."""
        question, answer = content
        content = f"Q: {question}\nA: {answer}"
        meta_data = {
            "url": "local",
        }
        return [
            {
                "content": content,
                "meta_data": meta_data,
            }
        ]

File: embedchain-js/embedchain/loaders/LocalQnaPair.ts
import type { LoaderResult, QnaPair } from '../models';
import { BaseLoader } from './BaseLoader';

class LocalQnaPairLoader extends BaseLoader {
  // eslint-disable-next-line class-methods-use-this
  async loadData(content: QnaPair): Promise<LoaderResult> {
    const [question, answer] = content;
    const contentText = `Q: ${question}\nA: ${answer}`;
    const metaData = {
      url: 'local',
    };
    return [
      {
        content: contentText,
        metaData,
      },
    ];
  }
}

export { LocalQnaPairLoader };
